### PR TITLE
Add unit tests for internal.databases.greenplum.buildFetchCommand

### PR DESCRIPTION
**Pull request by denisSurkov**
# Pull request description

### Describe what this PR fix
Adding unit tests for internal.databases.greenplum.buildFetchCommand

### CODE REVIEW

File: internal/databases/greenplum/backup_fetch_handler_test.go [hunk 1]
```diff
@@ -50,3 +51,131 @@ func TestPrepareContentIDsToFetch(t *testing.T) {
...
+	}()
+	config.CfgFile = "testConfig"
+
+	testcases := []struct {
+		handler   *FetchHandler
+		contentID int
+		cmdLine   string
+	}{
+		{
+			&FetchHandler{
```
Comment (denisSurkov) [inline internal/databases/greenplum/backup_fetch_handler_test.go:68]: I'm not sure if I should use `NewFetchHandler` or construct an object like this. At first I tried to use `NewFetchHandler` but it quicky became very complicated.

But I guess I should use `NewFetchHandler`. It's easer to refactor. I think no one will be happy if they change `cluster.SeqConfig` object and this test will fail. 

But I need some tips how to call `NewFetchHandler` correctly without a headache.

File: internal/databases/greenplum/backup_fetch_handler_test.go [hunk 2]
```diff
@@ -50,3 +51,131 @@ func TestPrepareContentIDsToFetch(t *testing.T) {
 		assert.Equal(t, tc.contentIDsToFetch, contentIDsToFetch)
 	}
 }
+
+func TestBuildFetchCommand(t *testing.T) {
+	beforeValue := config.CfgFile
+	defer func() {
+		config.CfgFile = beforeValue
+	}()
+	config.CfgFile = "testConfig"
```
Comment (denisSurkov) [inline internal/databases/greenplum/backup_fetch_handler_test.go:60]: I agree this is tricky. But `buildFetchCommand` relies on this global variable. May be there is some other way to do it. 

Without this setting `cmdLine` would be something like this `... --config= ...`. I think it's not good.

**Comment by denisSurkov** [comment]
@usernamedt , could you help with review?

**Comment by denisSurkov** [comment]
@x4m , sorry to bother you, but may be you could help with review too?

File: internal/databases/greenplum/backup_fetch_handler_test.go (hunk 1 above)
Comment (debebantur) [reply internal/databases/greenplum/backup_fetch_handler_test.go:68]: Well, if you want to use `NewFetchHandler `, the most difficult things seem to be `BackupSentinelDto` and `SegConfigMaker`. From the first one you need only segments, so you can define it like
```
dto := BackupSentinelDto{
		Segments: []SegmentMetadata{
			{
			DatabaseID: 1,
			ContentID: 1,
			Role: Primary,
			Port: 1234,
			Hostname: "name",
			DataDir: "datadir",		
			BackupID: "backup",
			},
		},
	}
```
And `SegConfigMaker` you can make from string:
```
seg, err := NewRestoreCfgSegMaker(strings.NewReader(`{"segments":{"1":{"hostname":"name", "port":1234, "data_dir": "dir"}}}`))
```

Comment (denisSurkov) [reply internal/databases/greenplum/backup_fetch_handler_test.go:68]: @debebantur , thank you

> if you want to use

Do you think I should do it the way you suggested or it's okey to leave it like this?

And with machinery all this variables is it possible to declare it in `testcases` array?

Comment (debebantur) [reply internal/databases/greenplum/backup_fetch_handler_test.go:68]: I believe that it is ok to leave it, anyway we need to test only `buildFetchCommand`. Also it seems difficult to declare all this structures in testcases array.

File: internal/databases/greenplum/backup_fetch_handler_test.go [hunk 3]
```diff
@@ -50,3 +51,131 @@ func TestPrepareContentIDsToFetch(t *testing.T) {
...
+				"--target-user-data=\"{\\\"id\\\":\\\"other-value-from-testing\\\"}\" " +
+				"--config=testConfig " +
+				"--restore-only=test1,test2 " +
+				">> /wal-g-log-seg1.log 2>&1",
+		},
+	}
+
+	for _, tc := range testcases {
+		cmdLine := tc.handler.buildFetchCommand(tc.contentID)
+		assert.Equal(t, tc.cmdLine, cmdLine)
```
[the remark this answers is not in the document]
Comment (debebantur) [reply internal/databases/greenplum/backup_fetch_handler_test.go:179]: Sorry, i forgot that `Fatalf` does not throw panic but instead exits process. Still we can test that we do not have backup id for content id. 
We can write something like 
```
func TestBuildFetchCommandCrushes(t *testing.T) {
	handler := &FetchHandler{
		cluster: &cluster.Cluster{
			ByContent: map[int][]*cluster.SegConfig{
				1: {
					{
						DbID:      1,
						ContentID: 2,
						Role:      "controlled",
						Port:      1234,
						Hostname:  "test.com",
						DataDir:   "/etc/test/",
					},
				},
			},
		},
		backupIDByContentID: map[int]string{},
		backup:              internal.Backup{},
		contentIDsToFetch:   map[int]bool{1: true},
	}

	if os.Getenv("TEST_BUILD_FETCH_COMMAND") == "1" {
		handler.buildFetchCommand(1)
		return
	}

	cmd := exec.Command(os.Args[0], "-test.run=TestBuildFetchCommandCrushes")
	cmd.Env = append(os.Environ(), "TEST_BUILD_FETCH_COMMAND=1")
	err := cmd.Run()
	if e, ok := err.(*exec.ExitError); ok && !e.Success() {
		return
	}
	t.Fatalf("process ran with err %v, want exit status 1", err)
}
```

[the remark this answers is not in the document]
Comment (denisSurkov) [reply internal/databases/greenplum/backup_fetch_handler_test.go:181]: That's actually pretty clever, thank you!

**Comment by denisSurkov** [comment]
@debebantur , so, I guess that's it

Thank you for your time and patience

Of course, If you have something else to mention I will fix it

But if everything looks good than what should I do next? 

Wait until someone merges it? May be you know somebody to ask to merge?

File: internal/databases/greenplum/backup_fetch_handler_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (debebantur) [reply internal/databases/greenplum/backup_fetch_handler_test.go:181]: Yep, that`s correct. If you want more cool ideas, there is a presentation by one of go developers https://go.dev/talks/2014/testing.slide#1

**Comment by debebantur** [comment]
> @debebantur , so, I guess that's it
> 
> Thank you for your time and patience
> 
> Of course, If you have something else to mention I will fix it
> 
> But if everything looks good than what should I do next?
> 
> Wait until someone merges it? May be you know somebody to ask to merge?

LGTM, we are merging it

**Review by debebantur — APPROVED** [review]

**Pull request merged**